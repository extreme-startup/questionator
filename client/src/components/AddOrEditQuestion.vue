<template>
  <Modal @close="close">
    <template v-slot:header>
      <Title>{{ titles.title }}</Title>
    </template>
    <QuestionForm
      :question="question"
      :submitTitle="titles.submitTitle"
      :errors="errors"
      :editFieldsConfig="modalType === 'edit' ? editFieldsConfig : null"
      @submit="addOrEditQuestion"
      @close="close"
    ></QuestionForm>
  </Modal>
</template>

<script>
import styled from 'vue-styled-components';
import Modal from '@/common/Modal.vue';
import QuestionForm from '@/components/QuestionForm.vue';
import { TitleH3 } from '@/common/styledComponents';

const Title = styled(TitleH3)`
  width: 100%;
  padding: 30px 0 30px 60px;
  margin: 0;
  text-align: left;
  font-weight: 600;
  box-sizing: border-box;
  background-color: var(--bg-color);
`;

const getModalTitles = modalType => {
  switch (modalType) {
    case 'add':
      return { title: 'Add new question', submitTitle: 'Save' };
    case 'edit':
      return { title: 'Edit question', submitTitle: 'Update' };
    default:
      return {};
  }
};

const editFieldsConfig = ['answer', 'value'];

export default {
  name: 'addOrEditQuestion',
  data: () => {
    return {
      errors: {
        text: {
          required: false,
        },
        answer: {
          required: false,
        },
      },
      editFieldsConfig,
    };
  },
  props: ['isShown', 'question', 'modalType'],
  components: {
    Modal,
    Title,
    QuestionForm,
  },
  computed: {
    titles: function() {
      return getModalTitles(this.modalType);
    },
  },
  methods: {
    close() {
      this.errors = {
        text: {
          required: false,
        },
        answer: {
          required: false,
        },
      };

      this.$store.dispatch('form/hideForm');
    },
    addOrEditQuestion(question) {
      if (this.modalType === 'add') {
        this.$store.dispatch('question/addQuestion', question);
      }

      if (this.modalType === 'edit') {
        this.$store.dispatch('question/updateQuestion', question);
      }

      this.close();
    },
  },
};
</script>
