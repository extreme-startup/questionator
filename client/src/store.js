import Vue from 'vue';
import Vuex from 'vuex';

Vue.use(Vuex);

export default new Vuex.Store({
  state: {
    user: null,
    isUserLoggedIn: false,
  },
  mutations: {
    setUser(state, user) {
      state.user = user;
      state.isUserLoggedIn = !!user;
    },
  },
  actions: {
    login({ commit }) {},
    setUser({ commit }, user) {
      commit('setUser', user);
    },
  },
});
