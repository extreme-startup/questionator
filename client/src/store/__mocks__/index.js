import Vue from 'vue';
import Vuex from 'vuex';

Vue.use(Vuex);

// common
export const getters = {};
export const mutations = {
  setUser: jest.fn(),
};
export const actions = {
  login: jest.fn(),
  setUser: jest.fn(),
};
export const state = {
  user: null,
  isUserLoggedIn: false,
};

// form
const defaults = {
  type: 'static',
  text: '',
  answer: '',
  value: 0,
};
export const formGetters = ({ isShown = true, modalType = 'edit', question = defaults }) => ({
  isShown: jest.fn().mockReturnValue(isShown),
  modalType: jest.fn().mockReturnValue(modalType),
  question: jest.fn().mockReturnValue(question),
});
export const formMutations = {
  show: jest.fn(),
  hide: jest.fn(),
  setType: jest.fn(),
  setQuestion: jest.fn(),
};
export const formActions = {
  showForm: jest.fn(),
  hideForm: jest.fn(),
};
export const formState = {
  isShown: false,
  modalType: null,
  question: { ...defaults },
};

export const form = {
  getters: formGetters({}),
  mutations: formMutations,
  actions: formActions,
  state: formState,
};

// question
export const questionState = {
  questions: {
    data: [],
    isFetching: false,
    error: null,
  },
  addQuestion: {
    isFetching: false,
    error: null,
  },
  updateQuestion: {
    isFetching: false,
    error: null,
  },
  deleteQuestion: {
    isFetching: false,
    error: null,
  },
};
const qs = [
  {
    id: '7737dahh-2b90-4e74-9f2f-b487c64556c6',
    type: 'static',
    text: 'test1',
    answer: 'scdscds',
    value: 4,
  },
  {
    id: '6737dace-2b90-5654-9f2f-b487c6455678',
    type: 'dynamic',
    text: 'test2',
    answer: 'ryrtty ghf vcxv',
    value: 5,
  },
];

export const questionGetters = ({ questions = qs }) => ({
  questions: jest.fn().mockReturnValue(questions),
  questionsFetchingStatus: jest.fn().mockReturnValue({ isFetching: false, error: null }),
  addQuestionFetchingStatus: jest.fn().mockReturnValue({ isFetching: false, error: null }),
  updateQuestionFetchingStatus: jest.fn().mockReturnValue({ isFetching: false, error: null }),
  deleteQuestionFetchingStatus: jest.fn().mockReturnValue({ isFetching: false, error: null }),
});

export const questionMutations = {
  setQuestions: jest.fn(),
  setQuestionsIsFetching: jest.fn(),
  setQuestionsError: jest.fn(),
  saveQuestion: jest.fn(),
  saveQuestionIsFetching: jest.fn(),
  saveQuestionError: jest.fn(),
  updateQuestion: jest.fn(),
  updateQuestionIsFetching: jest.fn(),
  updateQuestionError: jest.fn(),
  deleteQuestion: jest.fn(),
  deleteQuestionIsFetching: jest.fn(),
  deleteQuestionError: jest.fn(),
};

export const questionActions = {
  getQuestions: jest.fn(),
  addQuestion: jest.fn(),
  updateQuestion: jest.fn(),
  deleteQuestion: jest.fn(),
};

export const question = {
  state: questionState,
  getters: questionGetters({}),
  mutations: questionMutations,
  actions: questionActions,
};

export function __createMocks(custom = { getters: {}, mutations: {}, actions: {}, state: {} }) {
  const mockGetters = Object.assign({}, getters, custom.getters);
  const mockMutations = Object.assign({}, mutations, custom.mutations);
  const mockActions = Object.assign({}, actions, custom.actions);
  const mockState = Object.assign({}, state, custom.state);

  const mockedModules = {
    question,
    form,
  };

  return {
    getters: mockGetters,
    mutations: mockMutations,
    actions: mockActions,
    state: mockState,
    modules: mockedModules,
    store: new Vuex.Store({
      getters: mockGetters,
      mutations: mockMutations,
      actions: mockActions,
      state: mockState,
      modules: mockedModules,
    }),
  };
}

export const store = __createMocks().store;
