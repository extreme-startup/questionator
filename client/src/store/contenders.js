import * as api from '@/contenders/api';

const CUSTOMERS_POLLING_INTERVAL = 2000;

const contendersState = {
  contenders: [],
  isFetching: false,
  error: null,
  isGetContendersPolling: {
    timerId: null,
    isStarted: false,
    error: null,
  },
};

const getters = {
  getContenders: state => state.contenders,
  getContendersFetchingStatus: state => {
    const { isFetching, error } = state;
    return { isFetching, error };
  },
  getIsGetContendersPollingStarted: state => {
    return state.isGetContendersPolling.isStarted;
  },
};

const contendersMutations = {
  setContenders(state, contenders) {
    state.contenders = contenders;
  },
  toggleContendersIsFetching(state) {
    state.contenders.isFetching = !state.contenders.isFetching;
  },
  setContendersError(state, error) {
    state.contenders.error = error;
  },
};

const pollingMutations = {
  setPollingTimerId(state, payload) {
    state.isGetContendersPolling.timerId = payload;
  },
  togglePollingIsStarted(state) {
    state.isGetContendersPolling.isStarted = !state.isGetContendersPolling.isStarted;
  },
  setPollingError(state, payload) {
    state.isGetContendersPolling.error = payload;
  },
};

const actions = {
  getContenders: async (context, payload) => {
    if (context.state.isFetching) {
      return;
    }
    context.commit('toggleContendersIsFetching');
    try {
      const { data } = await api.getContenders(payload);
      context.commit('setContenders', data);
    } catch (e) {
      context.commit('setContendersError', e);
    } finally {
      context.commit('toggleContendersIsFetching');
    }
  },
  startPollingContenders: async (context, payload) => {
    context.commit('togglePollingIsStarted');
    try {
      const pollingTimerId = setInterval(() => {
        context.dispatch('getContenders', payload);
      }, CUSTOMERS_POLLING_INTERVAL);
      context.commit('setPollingTimerId', pollingTimerId);
    } catch (err) {
      context.commit('setPollingError', err);
      context.dispatch('stopPollingContenders');
    }
  },
  stopPollingContenders: context => {
    clearInterval(context.state.isGetContendersPolling.timerId);
    context.commit('togglePollingIsStarted');
  },
};

const mutations = {
  ...contendersMutations,
  ...pollingMutations,
};

export default {
  namespaced: true,
  state: contendersState,
  getters,
  mutations,
  actions,
};
