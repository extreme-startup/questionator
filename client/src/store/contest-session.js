import * as api from '@/contest-session/api';
import router from '../router';

const sessionState = {
  sessions: {
    data: [],
    isFetching: false,
    error: null,
  },
  addSession: {
    isFetching: false,
    error: null,
  },
};

const getters = {
  sessions: state => state.sessions.data,
  sessionsFetchingStatus: state => ({
    isFetching: state.sessions.isFetching,
    error: state.sessions.error,
  }),
  addSessionFetchingStatus: state => ({
    isFetching: state.addSession,
  }),
};

const setSessionMutations = {
  setSessions(state, sessions) {
    state.sessions.data = sessions;
  },
  setSessionsIsFetching(state) {
    state.sessions.isFetching = !state.sessions.isFetching;
  },
  setSessionsError(state, error) {
    state.sessions.error = error;
  },
};

const addSessionMutation = {
  addSessionIsFetching(state) {
    state.addSession.isFetching = !state.addSession.isFetching;
  },
  addSessionError(state, error) {
    state.addSession.error = error;
  },
};

const mutations = {
  ...setSessionMutations,
  ...addSessionMutation,
};

const actions = {
  getSessions: async (context, payload) => {
    context.commit('setSessionsIsFetching');
    try {
      const { data } = await api.getSession(payload);
      context.commit('setSessions', data);
    } catch (e) {
      context.commit('setSessionsError', e);
    } finally {
      context.commit('setSessionsIsFetching');
    }
  },
  addSession: async (context, payload) => {
    context.commit('addSessionIsFetching');
    try {
      const session = await api.addSession(payload);
      router.push(`/training-session/${session.data.id}`);
      context.dispatch('getSessions', payload);
    } catch (e) {
      context.commit('addSessionError', e);
    } finally {
      context.commit('addSessionIsFetching');
    }
  },
};

export default {
  namespaced: true,
  state: sessionState,
  getters,
  mutations,
  actions,
};
