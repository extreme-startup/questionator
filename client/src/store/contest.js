import Vue from 'vue';
import * as api from '@/api/contest';

const contestState = {
  contenders: {},
  answeredQuestions: {
    data: {},
    lastUpdate: 0,
  },
  accumulatedAnsweredQuestions: {
    data: {},
  },
};

const getters = {
  answeredQuestions: state => state.answeredQuestions.data,
  accumulatedAnsweredQuestions: state => {
    const data = state.accumulatedAnsweredQuestions.data;
    return Object.keys(data).map(key => {
      return {
        label: key,
        borderColor: data[key].color,
        backgroundColor: data[key].color,
        data: data[key].answers,
      };
    });
  },
  lastUpdate: state => state.answeredQuestions.lastUpdate,
  // acumulated: state => {
  //   const answeredQuestions = state.answeredQuestions.data;
  //   const ret = Object.keys(answeredQuestions).map(key => {
  //     let color = getRandomColor();
  //     return {
  //       label: key,
  //       borderColor: color,
  //       backgroundColor: color,
  //       data: answeredQuestions[key].answers.reduce((acc, curr) => {
  //         if (!acc.length) {
  //           return acc.concat({ x: curr.answerTime, y: curr.score });
  //         }

  //         const aggregatedScore = acc[acc.length - 1].y + curr.score;

  //         return acc.concat({ x: curr.answerTime, y: aggregatedScore });
  //       }, []),
  //     };
  //   });
  //   let legend = Vue._.orderBy(
  //     ret.map(r => {
  //       return {
  //         username: r.label,
  //         score: r.data[r.data.length - 1].y,
  //       };
  //     }),
  //     'score',
  //     'desc',
  //   );

  //   return {
  //     datasets: ret,
  //     legend,
  //   };
  // },
};

function getRandomColor() {
  var letters = '0123456789ABCDEF'.split('');
  var color = '#';
  for (var i = 0; i < 6; i++) {
    color += letters[Math.floor(Math.random() * 16)];
  }
  return color;
}

const setAnsweredQuestionsMutations = {
  setAnsweredQuestions(state, answeredQuestions) {
    state.answeredQuestions.data = joinData(answeredQuestions, state.answeredQuestions.data);
  },
  setAccumulatedAnsweredQuestions(state, answeredQuestions) {
    state.accumulatedAnsweredQuestions.data = accumulateData(
      answeredQuestions,
      state.accumulatedAnsweredQuestions.data,
    );
  },
  increaseLastUpdate(state, lastUpdate) {
    state.answeredQuestions.lastUpdate = lastUpdate;
  },
};

const mutations = {
  ...setAnsweredQuestionsMutations,
};

function joinData(inputData, currData = {}) {
  return Object.keys(inputData).reduce((acc, curr) => {
    if (acc[curr]) {
      return {
        ...acc,
        [curr]: { answers: acc[curr].answers.concat(...inputData[curr]) },
      };
    } else {
      return {
        ...acc,
        [curr]: { answers: inputData[curr] },
      };
    }
  }, currData);
}

function accumulateData(inputData, currData = {}) {
  return Object.keys(inputData).reduce((acc, curr) => {
    if (acc[curr]) {
      let t1 = acc[curr];
      return {
        ...acc,
        [curr]: {
          color: t1.color,
          answers: inputData[curr].reduce((acc1, curr1) => {
            if (!acc1.length) {
              return acc1.concat({ x: curr1.answerTime, y: curr1.score });
            }

            const aggregatedScore = acc1[acc1.length - 1].y + curr1.score;

            return acc1.concat({ x: curr1.answerTime, y: aggregatedScore });
          }, t1.answers),
        },
      };
    } else {
      const color = getRandomColor();
      return {
        ...acc,
        [curr]: {
          color,
          answers: inputData[curr].reduce((answers, answer) => {
            if (!answers.length) {
              return answers.concat({ x: answer.answerTime, y: answer.score });
            }

            const aggregatedScore = answers[answers.length - 1].y + answer.score;

            return answers.concat({ x: answer.answerTime, y: aggregatedScore });
          }, []),
        },
      };
    }
  }, currData);
}

const actions = {
  getAnsweredQuestions: async ({ commit, state }) => {
    const { data } = await api.getAnsweredQuestions(null, state.answeredQuestions.lastUpdate);
    const groupped = Vue._.groupBy(data, 'userid');

    commit('setAccumulatedAnsweredQuestions', groupped);
    // commit('setAnsweredQuestions', groupped);
    commit('increaseLastUpdate', state.answeredQuestions.lastUpdate + 1);
  },
};

export default {
  namespaced: true,
  state: contestState,
  getters,
  mutations,
  actions,
};
