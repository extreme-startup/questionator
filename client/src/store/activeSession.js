import * as api from '@/contest-session/api';

const ACTIVE_SESSION_POLLING_INTERVAL = 2000;

const sessionStatus = {
  CREATED: 'created',
  IN_PROGRESS: 'in progress',
  PAUSED: 'paused',
  COMPLETED: 'completed',
};

const activeSessionState = {
  data: {
    players: [],
    trainer: null,
    startedTime: '',
    sessionHash: '',
    status: '',
    askedQuestion: [],
  },
  isFetching: false,
  error: null,
  isGetActiveSessionPolling: {
    timerId: null,
    isStarted: false,
    error: null,
  },
};

const getters = {
  getActiveSession: state => ({ ...state.data }),
  getActiveSessionFetchingStatus: state => {
    const { isFetching, error } = state;
    return { isFetching, error };
  },
  getIsActiveSessionPollingStarted: state => {
    return state.isGetActiveSessionPolling.isStarted;
  },
  getIsSessionStarted: state => {
    return state.data.status === sessionStatus.IN_PROGRESS;
  },
  getIsSessionPaused: state => {
    return state.data.status === sessionStatus.PAUSED;
  },
  getIsSessionCompleted: state => {
    return state.data.status === sessionStatus.COMPLETED;
  },
};

const activeSessionMutations = {
  setActiveSession(state, activeSession) {
    state.data = activeSession;
  },
  toggleActiveSessionIsFetching(state) {
    state.isFetching = !state.isFetching;
  },
  setActiveSessionError(state, error) {
    state.error = error;
  },
  startActiveSession(state) {
    state.data.startedTime = new Date();
    state.data.status = sessionStatus.IN_PROGRESS;
  },
  pauseActiveSession(state) {
    state.data.status = sessionStatus.PAUSED;
  },
};

const pollingMutations = {
  setPollingTimerId(state, payload) {
    state.isGetActiveSessionPolling.timerId = payload;
  },
  togglePollingIsStarted(state) {
    state.isGetActiveSessionPolling.isStarted = !state.isGetActiveSessionPolling.isStarted;
  },
  setPollingError(state, payload) {
    state.isGetActiveSessionPolling.error = payload;
  },
};

const actions = {
  getActiveSession: async (context, payload) => {
    if (context.state.isFetching) {
      return;
    }
    context.commit('toggleActiveSessionIsFetching');
    try {
      const { data: activeSession } = await api.getActiveSession(payload);
      context.commit('setActiveSession', activeSession);
    } catch (e) {
      context.commit('setActiveSessionError', e);
    } finally {
      context.commit('toggleActiveSessionIsFetching');
    }
  },
  startPollingActiveSession: async (context, payload) => {
    context.commit('togglePollingIsStarted');
    try {
      const pollingTimerId = setInterval(() => {
        context.dispatch('getActiveSession', payload);
      }, ACTIVE_SESSION_POLLING_INTERVAL);
      context.commit('setPollingTimerId', pollingTimerId);
    } catch (err) {
      context.commit('setPollingError', err);
      context.dispatch('stopPollingActiveSession');
    }
  },
  stopPollingActiveSession: context => {
    clearInterval(context.state.isGetActiveSessionPolling.timerId);
    context.commit('togglePollingIsStarted');
  },
  updateSession: async (context, payload) => {
    try {
      await api.updateSession(context.state.data);
    } catch (err) {
      context.dispatch('setActiveSessionError', err);
    }
  },
  startActiveSession: async context => {
    try {
      await api.startSession({ id: context.state.data.id });
      context.commit('startActiveSession');
    } catch (e) {
      context.dispatch('setActiveSessionError', e);
    }
  },
  pauseActiveSession: async context => {
    try {
      await api.pauseSession({ id: context.state.data.id });
      context.commit('pauseActiveSession');
    } catch (e) {
      context.dispatch('setActiveSessionError', e);
    }
  },
  continueActiveSession: async context => {
    try {
      await api.continueSession({ id: context.state.data.id });
      context.commit('startActiveSession');
    } catch (e) {
      context.dispatch('setActiveSessionError', e);
    }
  },
  stopActiveSession: async context => {
    try {
      await api.completeSession({ id: context.state.data.id });
      context.commit('stopActiveSession');
    } catch (e) {
      context.dispatch('setActiveSessionError', e);
    }
  },
};

const mutations = {
  ...activeSessionMutations,
  ...pollingMutations,
};

export default {
  namespaced: true,
  state: activeSessionState,
  getters,
  mutations,
  actions,
};
