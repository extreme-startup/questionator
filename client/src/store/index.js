import Vue from 'vue';
import Vuex from 'vuex';
import question from './question';
import contestSession from './contest-session';
import form from './form';
import contest from './contest';
import contenders from './contenders';
import activeSession from './activeSession';

Vue.use(Vuex);

const store = new Vuex.Store({
  state: {
    user: null,
    isUserLoggedIn: false,
  },
  getters: {
    userId: state => state.user.id,
  },
  mutations: {
    setUser(state, user) {
      state.user = user;
      state.isUserLoggedIn = !!user;
    },
  },
  actions: {
    login({ commit }) {},
    setUser({ commit }, user) {
      commit('setUser', user);
    },
  },
  modules: {
    question,
    form,
    contest,
    contestSession,
    contenders,
    activeSession,
  },
});

export default store;
