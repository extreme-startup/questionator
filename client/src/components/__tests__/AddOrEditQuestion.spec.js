import { mount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import { __createMocks as createStoreMocks } from '../../store';
import AddOrEditQuestion from '../AddOrEditQuestion.vue';

jest.mock('../../store');

const localVue = createLocalVue();
localVue.use(Vuex);

const question = {
  type: 'static',
  text: '',
  answer: '',
  value: 0,
};

describe('AddOrEditQuestion', () => {
  let wrapper;
  let storeMocks;

  beforeEach(() => {
    storeMocks = createStoreMocks();

    wrapper = mount(AddOrEditQuestion, {
      store: storeMocks.store,
      localVue,
      propsData: {
        modalType: 'add',
        question,
      },
    });
  });

  test('renders add form correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('renders edit form correctly', () => {
    wrapper = mount(AddOrEditQuestion, {
      store: storeMocks.store,
      localVue,
      propsData: {
        modalType: 'edit',
        question,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
