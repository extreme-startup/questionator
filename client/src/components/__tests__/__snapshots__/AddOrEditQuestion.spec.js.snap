// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddOrEditQuestion renders add form correctly 1`] = `
<div
  class="hZfSeG"
  name="modal-fade"
>
  <div
    class="kuljaD"
  >
    <header>
      <button
        class="gsyLbm"
      >
        x
      </button>
       
      <h1
        class="ePoQLf fdLKPL jBBvdc"
      >
        Add new question
      </h1>
    </header>
     
    <section>
       
      <div
        class="bjGaHK"
      >
        <form
          novalidate="novalidate"
        >
          <div
            class="jtUfTw kqXKrG"
          >
            <button
              class="dbZTLp"
            >
              
        Static
      
            </button>
             
            <button
              class="kJsYHB"
            >
              
        Dynamic
      
            </button>
          </div>
           
          <!---->
           
          <div
            class="jtUfTw"
          >
            <input
              class="formInput"
              id="question-text-input"
              name="questionText"
              placeholder="Type your question"
              required="required"
              type="text"
            />
             
            <!---->
             
            <!---->
          </div>
           
          <div
            class="jtUfTw"
          >
            <textarea
              class="formTextArea"
              id="question-answer-input"
              name="answer"
              placeholder="Type your answer"
              required="required"
            />
             
            <!---->
             
            <!---->
          </div>
           
          <div
            class="jtUfTw"
          >
            <input
              class="formInput"
              id="question-value-input"
              name="questionValue"
              placeholder="Enter question value (points)"
              required="required"
              type="text"
            />
             
            <!---->
             
            <!---->
          </div>
           
          <div
            class="fvVliX"
          >
            <button
              class="jYfsmY liYJHj"
              id="question-cancel-button"
              type="reset"
            >
              
        Cancel
      
            </button>
             
            <button
              class="lfvxis liYJHj"
              id="question-save-button"
              type="submit"
            >
              Save
            </button>
          </div>
        </form>
      </div>
    </section>
     
    <footer />
  </div>
</div>
`;

exports[`AddOrEditQuestion renders edit form correctly 1`] = `
<div
  class="hZfSeG"
  name="modal-fade"
>
  <div
    class="kuljaD"
  >
    <header>
      <button
        class="gsyLbm"
      >
        x
      </button>
       
      <h1
        class="ePoQLf fdLKPL jBBvdc"
      >
        Edit question
      </h1>
    </header>
     
    <section>
       
      <div
        class="bjGaHK"
      >
        <form
          novalidate="novalidate"
        >
          <!---->
           
          <p
            class="gYaVnZ"
          >
            Type: static
          </p>
           
          <div
            class="jtUfTw"
          >
            <!---->
             
            <p
              class="gYaVnZ"
            >
              Question: 
            </p>
             
            <!---->
          </div>
           
          <div
            class="jtUfTw"
          >
            <textarea
              class="formTextArea"
              id="question-answer-input"
              name="answer"
              placeholder="Type your answer"
              required="required"
            />
             
            <!---->
             
            <!---->
          </div>
           
          <div
            class="jtUfTw"
          >
            <input
              class="formInput"
              id="question-value-input"
              name="questionValue"
              placeholder="Enter question value (points)"
              required="required"
              type="text"
            />
             
            <!---->
             
            <!---->
          </div>
           
          <div
            class="fvVliX"
          >
            <button
              class="jYfsmY liYJHj"
              id="question-cancel-button"
              type="reset"
            >
              
        Cancel
      
            </button>
             
            <button
              class="lfvxis liYJHj"
              id="question-save-button"
              type="submit"
            >
              Update
            </button>
          </div>
        </form>
      </div>
    </section>
     
    <footer />
  </div>
</div>
`;
